### DISCUSSION TITLE
How to achieve reactive behavior (auto-update stream) with background fetching

### DISCUSSION THREAD

**Discussion by rubenferreira97** [simolus3/drift#1057] [Q&A]
Currently I have all my queries being called on a singleton instance given by getIt like this:
`getIt.registerSingleton<AppDb>(AppDb());`

Now I want to accomplish a periodic background fetch every x minutes, and upsert data on the database. 
How should I approach this using a library like WorkManager (https://github.com/fluttercommunity/flutter_workmanager) or background_fetch (https://github.com/transistorsoft/flutter_background_fetch) with moor?

I have read the documentation but I couldn't see any example of this.
I fear I might lock the database since different isolates may call two different instances.

I want to be able to see the changes that might occur on the background fetch.

**Comment by simolus3** [comment]
> I want to be able to see the changes that might occur on the background fetch.

In real-time? If so, you could use the [moor isolates](https://moor.simonbinder.eu/docs/advanced-features/isolates/) library. It allows you to synchronize stream-queries between isolates.
I'm not exactly sure how the workmanager works under the hood. If it's spawning another Flutter Engine, it might not be possible to use real-time communication. I'm not sure what the best way to approach this would be in that case.
In the next moor update, we'll have the more general "remote"-api, which allows you to use moor over any two-way communication stream. So if you know of any way to communicate between the UI and a worker in real-time, I can show you how to use moor over that connection.
